### ISSUE TITLE
getParent type is incompatible with newest TS versions

### ISSUE THREAD

**Issue by Validark** [dsherret/ts-morph#692]
In the latest version of TypeScript, this will give you a compile-time error:
```ts
function foo(node: ts.Identifier | ts.ExpressionWithTypeArguments) {
	node.getParent();
}
```

The errors look something like this:
![image](https://user-images.githubusercontent.com/15217173/63656789-b8099600-c75e-11e9-9fcc-4588d644bdd2.png)

This happens due to a change in the way TypeScript handles method unions. If you think this error is incorrect, perhaps a TypeScript issue should be made?

**Comment by dsherret** [comment]
I haven't figured out a way to fix this yet and I'm not sure it's possible in TS 3.6 without changing the method to always being nullable.

This is due to the issue you outlined here: https://github.com/microsoft/TypeScript/issues/32506

**Comment by dsherret** [comment]
I'm so sad, but I'm probably going to do a code generation solution in the final declaration file to solve this :(

**Comment by dsherret** [comment]
Yeah, this solution is no fun, but I don't see any way around it. This will be fixed in version 4.0 (I hope to get a release out by Tuesday).
